src/data/validate.rs

```diff
@@ -0,0 +1,365 @@
+use crate::config::Mapping;
+
+/// Validate paths for entity
+pub fn validate_paths_for_entity(
+    entity: &str,
+    mappings: &Vec<Mapping>,
+    api_schema: &serde_json::Map<String, serde_json::Value>,
+) -> anyhow::Result<()> {
+    // if entity name is not set in api_schema throw an exception
+    if !api_schema.contains_key(entity) {
+        anyhow::bail!("Entity {} not found in API schema", entity);
+    }
+
+    for entry in mappings {
+        let path_mapping = match entry {
+            Mapping::ByPath(path_mapping) => path_mapping,
+            Mapping::ByScript(_) => continue,
+        };
+
+        let path = path_mapping.entity_path.split('.').collect::<Vec<_>>();
+
+        // if path starts with ? its optional
+        if path[0].ends_with('?') {
+            continue;
+        }
+
+        // check if first path element is set in the entity object in schema
+        if !api_schema[entity]["properties"][path[0]].is_object() {
+            anyhow::bail!("Entity {} does not have a field {}", entity, path[0]);
+        }
+
+        let root_path = api_schema[entity]["properties"][path[0]].as_object().unwrap();
```

Comment:
More rust way of doing this:
```suggestion
        let root_path = match api_schema[entity]["properties"][path[0]].as_object() {
            Some(obj) => obj,
            None => {
                 anyhow::bail!("Entity {} does not have a field {}", entity, path[0]);
             }
        };
```
You could also write it like this
```suggestion
        let root_path = if let Some(obj) = api_schema[entity]["properties"][path[0]].as_object() {
            obj
        } else {
            anyhow::bail!("Entity {} does not have a field {}", entity, path[0]);
        };
```
Or even nicer:
```suggestion
        let Some(root_path) = api_schema[entity]["properties"][path[0]].as_object() else {
            anyhow::bail!("Entity {} does not have a field {}", entity, path[0]);
        };
```